### PR TITLE
8236176: Parallel GC SplitInfo comment should be updated for shadow regions

### PR DESCRIPTION
**Pull request by walulyai**
Hi all,

Please review this small change to add a high-level description of how shadow regions are utilized during the compact phase of the parallel GC. There is no special handling of Splitinfo by the shadow regions; the destination computation and the handling of partial_objects are done during the summary phase while the shadow regions optimization only applies to compaction. 

Additionally, some unused variables are cleaned up.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8236176](https://bugs.openjdk.java.net/browse/JDK-8236176): Parallel GC SplitInfo comment should be updated for shadow regions

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to 52fbb6fafd0ca2870c251be06193a344a1a30cdc
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5195/head:pull/5195` \
`$ git checkout pull/5195`

Update a local copy of the PR: \
`$ git checkout pull/5195` \
`$ git pull https://git.openjdk.java.net/jdk pull/5195/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5195`

View PR using the GUI difftool: \
`$ git pr show -t 5195`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5195.diff">https://git.openjdk.java.net/jdk/pull/5195.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/parallel/psParallelCompact.hpp [hunk 1]
```diff
@@ -961,6 +961,25 @@ inline void ParMarkBitMapClosure::decrement_words_remaining(size_t words) {
 // regions and regions compacting into themselves.  There is always at least 1
 // region that can be put on the ready list.  The regions are atomically added
 // and removed from the ready list.
+//
+// During compaction, there is a natural task dependency among regions because
+// destination regions may also be source regions themselves.  Consequently, the
+// destination regions are not available for processing until all live objects
+// within them are evacuated to their destinations. These dependencies lead to
```
Comment (tschatzl) [inline src/hotspot/share/gc/parallel/psParallelCompact.hpp:968]: There is a mix of one and two spaces after a full stop (5 times I think) in this and the next paragraph. I recommend using two spaces like in the above paragraphs for uniformity.

File: src/hotspot/share/gc/parallel/psParallelCompact.hpp [hunk 2]
```diff
@@ -961,6 +961,25 @@ inline void ParMarkBitMapClosure::decrement_words_remaining(size_t words) {
 // regions and regions compacting into themselves.  There is always at least 1
 // region that can be put on the ready list.  The regions are atomically added
 // and removed from the ready list.
+//
```
Comment (tschatzl) [inline src/hotspot/share/gc/parallel/psParallelCompact.hpp:964]: This commentactually refers to the sentence: "// An alternate strategy is being investigated for this deferral of updating." above.

This sentence should be removed as this offers no information about the current algorithm. Fwiw I do not know that actually anyone is working on that, and if so, a CR in JIRA would be more appropriate.

**Review by tschatzl — APPROVED** [review]
Lgtm apart from my comments about some comment typos.

File: src/hotspot/share/gc/parallel/psParallelCompact.hpp (hunk 2 above)
Comment (walulyai) [reply src/hotspot/share/gc/parallel/psParallelCompact.hpp:963]: Removed!

**Review by albertnetymk — APPROVED** [review]

**Comment by walulyai** [comment]
Thanks for the reviews!

/integrate

**Pull request closed without merging**